sdk/identity/Azure.Identity/tests/AuthorizationCodeCredentialTests.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Testing;
+using NUnit.Framework;
+
+namespace Azure.Identity.Tests
+{
+    public class AuthorizationCodeCredentialTests : ClientTestBase
+    {
+        public AuthorizationCodeCredentialTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        [Test]
+        public async Task AuthenticateWithAuthCodeMockAsync()
+        {
+            var expectedToken = Guid.NewGuid().ToString();
+            var authCode = Guid.NewGuid().ToString();
+            var clientId = Guid.NewGuid().ToString();
+            var tenantId = Guid.NewGuid().ToString();
+            var clientSecret = Guid.NewGuid().ToString();
+
+            MockResponse response = CreateAuthorizationResponse(expectedToken);
+
+            var mockTransport = new MockTransport(request => ProcessMockRequest(request, tenantId, expectedToken));
+
+            var options = new IdentityClientOptions() { Transport = mockTransport };
+
+            AuthorizationCodeCredential cred = InstrumentClient(new AuthorizationCodeCredential(tenantId, clientId, clientSecret, authCode, options));
+
+            AccessToken token = await cred.GetTokenAsync(new TokenRequest(new string[] { "https://vault.azure.net/.default" }));
```

Comment:
You call `GetTokenAsync` only once in tests, do you end up hitting the else code path with `AcquireTokenSilent`?

Reply:
Added a second call with a separate scope to exercise the AquireTokenSilent path